cgi-bin/LJ/Protocol.pm

```diff
@@ -1757,6 +1754,11 @@ sub postevent
         # TODO: error on failure?  depends on the job I suppose?  property of the job?
     }
 
+    # To minimize impact on legacy code, let's make sure the entry object in
+    # memory has been populated with data. Easiest way to do that is to call
+    # one of the methods that loads the relevant row from the database.
+    $entry->valid;
```

Comment:
Absolutely, the methods already do that, and if we used the methods everywhere, I wouldn't worry about it! But legacy stuff often uses hashrefs instead of methods - we saw the same thing with user objects early on.